### PR TITLE
SOV-617 add admin role in protocol

### PR DESCRIPTION
**Pull request by cwsnt**

### CODE REVIEW

File: tests/other/ProtocolSettings.test.js [hunk 1]
```diff
@@ -968,6 +968,76 @@ contract("ProtocolSettings", (accounts) => {
             expect(event["newRolloverFlexFeePercent"] == new_percent.toString()).to.be.true;
             expect((await sovryn.rolloverFlexFeePercent()).eq(new_percent)).to.be.true;
         });
+
+        it("Admin should not be able to set the admin address", async () => {
```
Comment (tjcloa) [inline tests/other/ProtocolSettings.test.js:972]: missing a test for a random address not owner and not admin not being able to set admin

**Review by tjcloa — CHANGES_REQUESTED** [review]
missing a test
the rest lgtm

File: tests/other/ProtocolSettings.test.js (hunk 1 above)
Comment (cwsnt) [reply tests/other/ProtocolSettings.test.js:972]: Updated in this [commit](https://github.com/DistributedCollective/Sovryn-smart-contracts/pull/511/commits/f185515e59597af142af64b9dfded7b23ec023e8)

File: contracts/modules/ProtocolSettings.sol [hunk 2]
```diff
@@ -393,6 +395,30 @@ contract ProtocolSettings is
...
+     *
+     * only owner can perform this action.
+     *
+     * @param newAdmin The new address of the admin.
+     * */
+    function setAdmin(address newAdmin) external onlyOwner {
+        address oldAdmin = admin;
+        admin = newAdmin;
+
+        emit SetAdmin(msg.sender, oldAdmin, newAdmin);
```
Comment (tjcloa) [inline contracts/modules/ProtocolSettings.sol:409]: emitting the event first eliminates need for a local variable oldAdmin and readin/writing

**Review by tjcloa — APPROVED** [review]
approved  
left a tiny recommendation in comments though

File: contracts/modules/ProtocolSettings.sol (hunk 2 above)
Comment (cwsnt) [reply contracts/modules/ProtocolSettings.sol]: updated in this [commit](https://github.com/DistributedCollective/Sovryn-smart-contracts/pull/511/commits/2c7cab319bc4d15bc3a796a3805f41b5786313e5)

**Pull request merged**